### PR TITLE
do not focus clients if a layer surface is focused

### PR DESCRIPTION
**Pull request by sevz17**
This fix an issue when you have a layer surface mapped (e.g. fuzzel) and you spawn various clients. You can use mod+j/mod+k to change its border color but not get real focus (the same happens with mouse input). The only surface should be focused is the layer surface

Do not merging until real world testing.

### CODE REVIEW

**Comment by fauxmight** [comment]
BEFORE committing this: popups from layer surface *do* appear, but do *not* show over XDG shell clients.
Example: hovering over waybar (layer shell) elements with *no* XDG shell clients on-screen *will* allow popups from waybar to appear.
Hovering over waybar (layer shell) elements when XDG shell clients *are* on screen -> popups are hidden by XDG shell clients.

AFTER committing this: *No* popups from layer shell clients appear.
Example: hovering over waybar elements *never* generates any popups.

File: dwl.c [hunk 1]
```diff
@@ -1606,7 +1611,7 @@ pointerfocus(Client *c, struct wlr_surface *surface, double sx, double sy,
 		focusclient(c, 0);
 
 	/* If surface is NULL, clear pointer focus */
-	if (!surface) {
+	if (!surface || exclusive_focus != surface) {
```
Comment (Sevz17) [inline dwl.c:1614]: @fauxmight can you revert this line and tell me if it works?

Comment (fauxmight) [reply dwl.c:1614]: No noticeable change.

**Comment by fauxmight** [comment]
Separate from the above issue: if I click on waybar (which normally *does* respond to mouse hover, but does *not* respond to mouse clicks), mouse *clicks* on XDG surfaces no longer register. Mouse *hover* still changes focus (and keyboard input) from one XDG client to another, but *clicks* do not. I can find no way to get mouse clicks' effects back to XDG clients.

Keyboard interaction is unaffected.

**Comment by sevz17** [comment]
And with the latest changes?

**Comment by fauxmight** [comment]
Now mouse clicks *never* register in any XDG shell client even if no layer shell client is started.

**Comment by sevz17** [comment]
@fauxmight, I forgot ask you again, how does this work on the present state?

**Pull request merged**